build/build_queryservice.sh

```diff
@@ -2,15 +2,21 @@
 set -e
 
 TARBALL="service-$QUERYSERVICE_VERSION-dist.tar.gz"
-TEMP_DIR="$(mktemp -d)"
+TARBALL_SIGNATURE="$TARBALL".md5
 
-wget "https://archiva.wikimedia.org/repository/releases/org/wikidata/query/rdf/service/$QUERYSERVICE_VERSION/$TARBALL" \
-    -O "$TEMP_DIR/$TARBALL"
+MD5_SIGNATURE="$(wget -qO- https://archiva.wikimedia.org/repository/releases/org/wikidata/query/rdf/service/$QUERYSERVICE_VERSION/$TARBALL_SIGNATURE)"
+CACHE_PATH="git_cache/queryservice/$MD5_SIGNATURE"
 
-TARBALL_PATH="$TEMP_DIR/$TARBALL"
+if [ ! -f "$CACHE_PATH/$TARBALL" ]; then
```

Comment:
Might be some good arguments for moving this to the update_cache.sh script as well rather than having this logic in the  build step? Not sure.

Reply:
Not sure… it seems that `update_cache.sh` only deals with git repos.

Reply:
alright, added a comment and lets keep it here.